packages/core/src/components/overlay/overlay.tsx

```diff
@@ -357,9 +359,6 @@ export class Overlay extends React.PureComponent<IOverlayProps, IOverlayState> {
         if (this.props.isOpen) {
             safeInvoke(this.props.didOpen);
         }
-        if (this.props.autoFocus) {
-            this.bringFocusInsideOverlay();
-        }
```

Comment:
not needed as `overlayWillOpen` is invoked shortly afterward for first render and then for every subsequent open, whereas this is only called after initial mount.